test/src/test/java/org/corfudb/integration/LogReplicationIT.java

```diff
@@ -811,16 +832,22 @@ public void testLogEntrySyncValidCrossTablesWithWritingAtSrc() throws Exception
         testConfig.clear();
         testConfig.setWritingSrc(true);
         testConfig.setDeleteOP(true);
-
         testConfig.setWaitOn(WAIT.ON_ACK);
-        startLogEntrySync(crossTables, WAIT.ON_ACK, false);
+
+        HashSet<WAIT> waitHashSet = new HashSet<>();
+        waitHashSet.add(WAIT.ON_ACK);
+        startLogEntrySync(crossTables, waitHashSet, true);
 
         expectedAckTimestamp = Long.MAX_VALUE;
-        // Verify Data on Destination site
-        System.out.println("****** Verify Data on Destination");
+
         // Because t2 is not specified as a replicated table, we should not see it on the destination
         srcDataForVerification.get(t2).clear();
 
+        // Verify Data on Destination site
+        System.out.println("****** Wait Data on Destination");
+        waitData(dstCorfuTables, srcDataForVerification);
```

Comment:
The intention of the observables and the WAIT.ON_ACK is to avoid while loops that are just stuck there until a condition is met. Can't we set the expected number of ACKS by knowing the size of what we wrote and how many messages will be sent, hence how many calks we will receive back?

Reply:
All you need is to set expectedAckMessages to the correct value.

Reply:
It is not accurate right. Even we know the value size, but not the MSR entry size in advance.